* grobian@gentoo.org: swab is found in stdlib on Solaris, so search
                      there or we get a redeclaration error, need C99 or
                      Solaris' headers will barf

--- configure
+++ configure
@@ -3618,23 +3618,6 @@
 echores "$_vsscanf"
 
 
-echocheck "swab()"
-cat > $TMPC << EOF
-#define _XOPEN_SOURCE 600
-#include <unistd.h>
-int main(void) { swab(0, 0, 0); return 0; }
-EOF
-_swab=no
-cc_check && _swab=yes
-if test "$_swab" = yes ; then
-  def_swab='#define HAVE_SWAB 1'
-  _need_swab=no
-else
-  def_swab='#undef HAVE_SWAB'
-  _need_swab=yes
-fi
-echores "$_swab"
-
 echocheck "POSIX select()"
 cat > $TMPC << EOF
 #include <stdio.h>
@@ -7931,6 +7913,26 @@
 
 cc_check -mno-omit-leaf-frame-pointer && cflags_no_omit_leaf_frame_pointer="-mno-omit-leaf-frame-pointer"
 
+# Solaris requires C99 semantics enabled when using this, hence this has
+# to go here after the CFLAGS trickery
+echocheck "swab()"
+cat > $TMPC << EOF
+#define _XOPEN_SOURCE 600
+#include <unistd.h>
+#include <stdlib.h>  /* Solaris */
+int main(void) { swab(0, 0, 0); return 0; }
+EOF
+_swab=no
+cc_check && _swab=yes
+if test "$_swab" = yes ; then
+  def_swab='#define HAVE_SWAB 1'
+  _need_swab=no
+else
+  def_swab='#undef HAVE_SWAB'
+  _need_swab=yes
+fi
+echores "$_swab"
+
 # This must be the last test to be performed. Any other tests following it
 # could fail due to linker errors. libdvdnavmini is intentionally not linked
 # against libdvdread (to permit MPlayer to use its own copy of the library).
